AppCenter/AppCenterSDK+HiDrive.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  AppCenterSDK+SMFLogger.swift
+//  SmartMobileFactory
+//
+//  Created by Konstantin Deichmann on 11.09.19.
+//  Copyright © 2019 SmartMobileFactory. All rights reserved.
+//
+
+import Foundation
+import AppCenter
+import AppCenterCrashes
+import AppCenterDistribute
+
+fileprivate enum AppCenterConstants {
+
+	static let appSecretKey					= "AppCenterAppSecret"
+}
+
+class AppCenterSDK: NSObject {
+
+	// MARK: - Private static properties
+
+	fileprivate static let isDebugBuild		: Bool = {
+		#if DEBUG
+		return true
+		#else
+		return false
+		#endif
+	}()
+
+	// MARK: - Private properties
+
+	fileprivate var isInitialized			= false
+	fileprivate var configuration			: Configuration?
+
+	// MARK: - Public properties
+
+	static var wasInitialized				: Bool {
+		return MSAppCenter.isConfigured()
+	}
+
+	// MARK: - Methods
+
+	/// This will setup the AppCenterSDK with the common base configuration. Crashes will be detected if the app is build with the release build type.
+	/// Distribution can be enabled using the configuration
+	///
+	/// - Parameters:
+	///   - configuration: Configuration object
+	static func setup(configuration: AppCenterSDK.Configuration = .default) {
+		guard (self.isDebugBuild == false || configuration.enableDebug == true) else {
+			return
+		}
+
+		let services = (configuration.isDistributionEnabled == true) ? [MSCrashes.self, MSDistribute.self] : [MSCrashes.self]
+
+		MSAppCenter.start(configuration.appSecret, withServices: services)
+		MSCrashes.setEnabled(HDManager.shared.settings.crashReportsEnabled)
+	}
+
+	/// Returns True, if and only if the Service got started and is enabled.
+	var isDistributionEnabled: Bool {
+		return MSDistribute.isEnabled()
+	}
+
+	/// Will enable or disable the Distribution Feature of AppCenter
+	/// While disabling works always.
+	/// For enabling the Setup (aka. start) - Method should be called before.
+	///
+	/// Flow in the App, for Apps that want to dynamically change that State:
+	/// - Call the Setup Method for with `isDistributionEnabled` set to true.
+	/// - Disable Distribution using `enableDistribution(enabled: false)`
+	/// - Enable Distribution at a later time using the same method
+	///
+	/// - Parameter enabled: Enable or Disable Distribtion
+	func enableDistribution(enabled: Bool = true) {
+		MSDistribute.setEnabled(enabled)
+	}
+
+	/// This will create a `fatalError` to crash the app.
+	static func performTestCrash() {
+
+		MSCrashes.generateTestCrash()
+	}
+
+	static let uiTestModeArgument        = "SMF.UITestMode"
+}
```

Comment:
Will https://smartmobilefactory.atlassian.net/browse/STRATODRVE-4127 be included in a later PR or has it been forgotten ?